hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/type/SqlLazyDeserializationTestUtil.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.sql.impl.type;
+
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.instance.AddressPicker;
+import com.hazelcast.instance.impl.DefaultNodeContext;
+import com.hazelcast.instance.impl.DefaultNodeExtension;
+import com.hazelcast.instance.impl.HazelcastInstanceFactory;
+import com.hazelcast.instance.impl.Node;
+import com.hazelcast.instance.impl.NodeContext;
+import com.hazelcast.instance.impl.NodeExtension;
+import com.hazelcast.internal.cluster.Joiner;
+import com.hazelcast.internal.server.Server;
+import com.hazelcast.internal.server.tcp.LocalAddressRegistry;
+import com.hazelcast.internal.server.tcp.ServerSocketRegistry;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.TestEnvironment;
+import com.hazelcast.internal.serialization.impl.InternalGenericRecord;
+
+import com.hazelcast.internal.serialization.InternalSerializationService;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+
+public class SqlLazyDeserializationTestUtil {
+    public static class SqlLazyDeserializationTestInstanceFactory extends TestHazelcastFactory {
```

Comment:
All of this is used in a single test, right? And, it is pretty specific to that test. What do you think about moving all of these definitions to that test?